client/my-sites/email/email-management/home/email-plan-warnings.tsx

```diff
@@ -19,18 +20,47 @@ type EmailPlanWarningsProps = {
 	emailAccount: EmailAccount;
 };
 
+const WARNING_CODE_OTHER_USER_OWNS_DOMAIN_SUBSCRIPTION = 'other-user-owns-subscription';
+const WARNING_CODE_OTHER_USER_OWNS_EMAIL = 'other-user-owns-email';
+
 const EmailPlanWarnings = ( { domain, emailAccount }: EmailPlanWarningsProps ) => {
 	const translate = useTranslate();
-	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteSlug = selectedSite?.slug ?? '';
 
 	const warning = emailAccount?.warnings?.[ 0 ];
 
 	if ( ! warning && canCurrentUserAddEmail( domain ) ) {
 		return null;
 	}
 
-	const cannotAddEmailReason = getCurrentUserCannotAddEmailReason( domain );
-	const cannotAddEmailMessage = cannotAddEmailReason?.message ?? '';
+	const cannotAddEmailWarningReason = getCurrentUserCannotAddEmailReason( domain );
+	const cannotAddEmailWarningMessage = cannotAddEmailWarningReason?.message ?? '';
+	const cannotAddEmailWarningCode = cannotAddEmailWarningReason?.code ?? null;
+
+	const WarningMessageForCode = ( props: { code: string | number | null } ) => {
+		switch ( props.code ) {
+			case WARNING_CODE_OTHER_USER_OWNS_EMAIL:
+				return (
+					<div className="email-plan-warnings__warning">
+						<div className="email-plan-warnings__message">
+							<span>{ cannotAddEmailWarningMessage }</span>
+						</div>
+					</div>
+				);
+			case WARNING_CODE_OTHER_USER_OWNS_DOMAIN_SUBSCRIPTION:
+				return (
+					<EmailNonDomainOwnerMessage
```

Comment:
We should be showing this message regardless of the error. If we're worried about the edge case when the user is a domain owner but not the subscription owner, we may need to update `useDomainOwnerUserName` so that it handles a query for getting email subscription owner as well. However, in reality, the owner of the domain should always be the owner of the email subscription. I'm inclined to not worry about that edge case. @fredrikekelund Thoughts?

Reply:
Well, it might seem an edge case, but every month a few support cases are indicating that it happens more often than we though :)